test/unit/dropbox.js

```diff
@@ -31,6 +39,40 @@ describe('Dropbox', () => {
       chai.assert.deepEqual({}, dbx.rpcRequest.getCall(0).args[1]);
     });
 
+    it('calls the correct request method (Download)', () => {
+      const dbx = new Dropbox();
+      const rpcSpy = sinon.spy(dbx, 'downloadRequest');
+      dbx.request('path', {}, USER_AUTH, 'api', DOWNLOAD)
+        .catch((error) => {
+          fail(error);
+        });
+      chai.assert.isTrue(rpcSpy.calledOnce);
+      chai.assert.equal('path', dbx.downloadRequest.getCall(0).args[0]);
+      chai.assert.deepEqual({}, dbx.downloadRequest.getCall(0).args[1]);
+    });
+
+    it('calls the correct request method (Upload)', () => {
+      const dbx = new Dropbox();
+      const rpcSpy = sinon.spy(dbx, 'uploadRequest');
+      dbx.request('path', {}, USER_AUTH, 'api', UPLOAD)
+        .catch((error) => {
+          fail(error);
+        });
+      chai.assert.isTrue(rpcSpy.calledOnce);
+      chai.assert.equal('path', dbx.uploadRequest.getCall(0).args[0]);
+      chai.assert.deepEqual({}, dbx.uploadRequest.getCall(0).args[1]);
+    });
+
+    it('throws an error for non-user auth (Upload)', () => {
+      const dbx = new Dropbox();
+      return chai.assert.isRejected(dbx.uploadRequest('path', {}, TEAM_AUTH, 'api'), Error, `Unexpected auth type: ${TEAM_AUTH}`);
+    });
+
+    it('throws an error for non-user auth (Download)', () => {
```

Comment:
Would it be cleaner to differentiate these into different test groups? Like making an upload suite, download suite, and an rpc suite?

Reply:
Yea I think that makes sense.